### PR TITLE
Reset option

### PR DESCRIPTION
**Pull request by bugsounet**
issue #3122

add reset with lockString options

---
Reset all lockStrings and set a new one in replacement

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -224,6 +224,11 @@ const MM = (function () {
 	 * @param {object} [options] Optional settings for the hide method.
 	 */
 	const hideModule = function (module, speed, callback, options = {}) {
+		// reset lockString and prepare for set the new one
+		if (options.reset === true && options.lockString) {
```
Comment (rejas) [inline js/main.js:228]: is checking for options.lockString necessary?

File: js/main.js [hunk 2]
```diff
@@ -268,6 +273,11 @@ const MM = (function () {
 	 * @param {object} [options] Optional settings for the show method.
 	 */
 	const showModule = function (module, speed, callback, options = {}) {
+		// reset lockString and prepare for set the new one
+		if (options.reset === true && options.lockString) {
```
Comment (rejas) [inline js/main.js:277]: I'd call the option "resetLockStrings" to make the usage clearer

**Review by rejas — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 1 above)
Comment (bugsounet) [reply js/main.js:228]: yes, because lockString replacement
reset and lockString works both (security)

File: js/main.js (hunk 2 above)
Comment (bugsounet) [reply js/main.js:277]: sure, You want to rename it or i do it ?

File: js/main.js (hunk 1 above)
Comment (bugsounet) [reply js/main.js:228]: I don't really see the point of destroying the Array without placing a new lock
but ... I may be wrong in my way of seeing things

@rejas: What do you think ?

**Comment by sdetweil** [comment]
but this is on hide, not show

explain the use case

**Comment by bugsounet** [comment]
it's on show too (I don't just think about me)

Personnaly with EXT-Pages:
 * Hide all modules on start (loading and initialized time) : i use `hide()`
 * reset lockString and set new lockString (like a manager)
 * Initialize time done: show needed module and timed rotate display

MMM-Page using:
 * Display all needed module in start (it use `show()`)
   * lockString are set by modules
   * conflit for display modules
 * first rotating time: strange and buggy (modules are displayed when it should be hidden)
 * rotating second time: all is clean

**Comment by rejas** [comment]
As I am not so deep into this issue, I would ask @sdetweil if he agrees with your usecase example.

**Comment by bugsounet** [comment]
Finally, humm... I Don't think really needed. It's just a MM² bug

Why ?

By reading not modified main code, I just understand this:

by using the callback fonction for hide() / show()  (From another module) AND if module updateDom() at the +/- same time
Callbacks function from another module will never executed. See [there](https://github.com/MichMich/MagicMirror/blob/develop/js/main.js#L310-L315)
maybe we have to recode hide/show / updateDom part

It's complex for reproduce but it's happen sometime

After, I can check/recode it. Before, I will Finish my AnimateCSS code 

So, in all case, I think we can close this PR

**Pull request closed without merging**